conda_devenv/devenv.py

```diff
@@ -210,14 +212,20 @@ def merge_dependencies_version_specifications(yaml_dict, key_to_merge):
                                    ' expected format.'.format(dep))
             # Consider the channel name as part of the package name. If multiple channels are specified, the package
             # will be repeated.
-            package_name = m.group('package')
-            if m.group('channel'):
-                package_name = m.group('channel') + package_name
+            if pip and ("+" in dep or ":" in dep):
```

Comment:
You could move the if statement to the top and avoid a unnecessary `re.match` and keep the "pip" handling isolated.

```python
if pip and ("+" in dep or ":" in dep):
    package_name = dep
    package_version = ''
else:
    m = re.match(package_pattern, dep)
    if m is None:
        raise RuntimeError('The package version specification "{}" do not follow the'
                           ' expected format.'.format(dep))
    # Consider the channel name as part of the package name. If multiple channels are specified, the package
    # will be repeated.
    package_name = m.group('package')
    if m.group('channel'):
        package_name = m.group('channel') + package_name

    package_version = m.group('version')
```

Also note that in this sample I did use ` package_version = ''` instead of an empty list.